senza/traffic.py

```diff
@@ -150,8 +150,18 @@ def set_new_weights(dns_names: list, identifier, lb_dns_name: str,
                                                "because traffic for it is 0".format(stack_name))
                 changed = True
                 continue
-            load_balancer = stack.template['Resources']['AppLoadBalancerMainDomain']
-            load_balancer['Properties']['Weight'] = percentage
+
+            for key, resource in stack.template['Resources'].items():
+                if (resource['Type'] == ResourceType.route53_record_set and
+                        resource['Properties']['Name'] == dns_name):
+                    load_balancer = stack.template['Resources'][key]
```

Comment:
can you change the variable name? `load_balancer` is misleading (as it's a Route53 record!)